### PR TITLE
Fix coverage and add to build

### LINKED ISSUE #323: fix code coverage
The `yarn coverage` commands in the `umbra-js/` and `contracts/` packages do not work

### PR DESCRIPTION
**Pull request by alexkeating**
### Description
- I fixed the existing coverage commands and added a monorepo command. I also added coverage to the github workflow

Closes #323

### CODE REVIEW

File: .github/workflows/ci.yaml
Comment (alexkeating) [inline .github/workflows/ci.yaml:73]: Build is failing because of an invalid mnemonic phrase is there something I need to do to setup the CI correctly? Things seem to be working locally

Comment (mds1) [reply .github/workflows/ci.yaml:73]: From a high level on scope here:
- Coverage for the `umbra-js` and `contracts-periphery` packages would be great
- We can skip `contracts-core` since the core contracts aren't changing anymore and that package isn't really touched these days
- Frontend coverage is roughly zero so lets skip that too

Then when testing the current state  of things it seems:
- Running `yarn coverage` in the `umbra-js` directory worked
- Running `forge coverage` in the `contracts-periphery` directory worked, so we probably need to add an alias into the package.json there

The mnemonic issue you mentioned above seems to be for the `contracts-core` directory but we can skip that for coverage.

`forge coverage --report lcov` creates a coverage report at `./lcov.info`. I'm sure `nyc` can gnerate LCOV too since that's a standard format. So then maybe what we do is generate the two LCOV reports, filter out certain directories (such as `test/` in `contracts-periphery`), merge the two files then check for a minimum overall coverage threshold in CI? Here is how we filter out directories and check for minimum coverage threshold for a pure forge repo: https://github.com/ScopeLift/foundry-template/blob/f2b8c5f72a4c85533957537e804e6816d9744bc8/.github/workflows/ci.yml#L42-L68

Comment (mds1) [reply .github/workflows/ci.yaml:73]: Ah wait but build is failing because of the missing mnemonic, so I just added one to github CI. I used `test test test test test test test test test test silent junk` based on this https://github.com/ScopeLift/umbra-protocol/blob/727d678ee30a42f656705b13bc7bb7904b3c36a3/contracts-core/scripts/deploy-registry.js#L58-L62

Comment (alexkeating) [reply .github/workflows/ci.yaml:74]: I think this should be working with lcov now.

File: contracts-periphery/Makefile
Comment (mds1) [inline contracts-periphery/Makefile:16]: If I run `yarn coverage` from the repo root, this part errors:

```
@umbra/contracts-periphery: Wrote LCOV report.
@umbra/contracts-periphery: sed: 1: "./lcov.info": invalid command code .
@umbra/contracts-periphery: make: *** [coverage] Error 1
@umbra/contracts-periphery: error Command failed with exit code 2.
@umbra/contracts-periphery: info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
lerna ERR! yarn run coverage exited 2 in '@umbra/contracts-periphery'
lerna WARN complete Waiting for 1 child process to exit. CTRL-C to exit immediately.
error Command failed with exit code 2.
```

IIRC sed behaves differently on macOS (which I'm using) vs linux, wonder if that's the error? Seems like it thinks the `.` part of `./lcov.info` is a command?

File: .github/workflows/ci.yaml
Comment (mds1) [reply .github/workflows/ci.yaml:74]: Awesome, yea all the CI checks now run so marking this resolved!

File: contracts-periphery/Makefile
Comment (alexkeating) [reply contracts-periphery/Makefile:16]: Ah that is probably the case, I tested on linux and I will resolve

Comment (alexkeating) [reply contracts-periphery/Makefile]: I think this should be resolved. I tested on mac and linux

Comment (mds1) [reply contracts-periphery/Makefile]: confirmed, thanks!

**Review by mds1 — APPROVED** [review]
LGTM, will defer to @apbendi for a second review before merging

**Comment by alexkeating** [comment]
@apbendi Bumping in case it was missed

**Review by apbendi — APPROVED** [review]
LGTM!

**Pull request merged**
